docs/visual-basic/programming-guide/concepts/linq/pre-atomization-of-xname-objects-linq-to-xml.md

```diff
@@ -4,83 +4,83 @@ ms.date: 07/20/2015
-Dim ID As XName = "ID"  
-  
-Dim root__2 As New XElement(Root__1, New XAttribute(XNamespace.Xmlns + "aw", aw), New XElement(Data, New XAttribute(ID, "1"), "4,100,000"), New XElement(Data, New XAttribute(ID, "2"), "3,700,000"), New XElement(Data, New XAttribute(ID, "3"), "1,150,000"))  
-  
-Console.WriteLine(root__2)  
-```  
-  
- This example produces the following output:  
-  
-```xml  
-<aw:Root xmlns:aw="http://www.adventure-works.com">  
-  <aw:Data ID="1">4,100,000</aw:Data>  
-  <aw:Data ID="2">3,700,000</aw:Data>  
-  <aw:Data ID="3">1,150,000</aw:Data>  
-</aw:Root>  
-```  
-  
- The following example is more similar to what you will likely encounter in the real world. In this example, the content of the element is supplied by a query:  
-  
-```vb  
-Dim Root__1 As XName = "Root"  
-Dim Data As XName = "Data"  
-Dim ID As XName = "ID"  
-  
-Dim t1 As DateTime = DateTime.Now  
-Dim root__2 As New XElement(Root__1, From i In System.Linq.Enumerable.Range(1, 100000)New XElement(Data, New XAttribute(ID, i), i * 5))  
-Dim t2 As DateTime = DateTime.Now  
-  
-Console.WriteLine("Time to construct:{0}", t2 - t1)  
-```  
-  
- The previous example performs better than the following example, in which names are not pre-atomized:  
-  
-```vb  
-Dim t1 As DateTime = DateTime.Now  
-Dim root As New XElement("Root", From i In System.Linq.Enumerable.Range(1, 100000)New XElement("Data", New XAttribute("ID", i), i * 5))  
-Dim t2 As DateTime = DateTime.Now  
-  
-Console.WriteLine("Time to construct:{0}", t2 - t1)  
+
+One way to improve performance in LINQ to XML is to pre-atomize <xref:System.Xml.Linq.XName> objects. Pre-atomization means that you assign a string to an <xref:System.Xml.Linq.XName> object before you create the XML tree by using the constructors of the <xref:System.Xml.Linq.XElement> and <xref:System.Xml.Linq.XAttribute> classes. Then, instead of passing a string to the constructor, which would use the implicit conversion from string to <xref:System.Xml.Linq.XName>, you pass the initialized <xref:System.Xml.Linq.XName> object.
+
+This improves performance when you create a large XML tree in which specific names are repeated. To do this, you declare and initialize <xref:System.Xml.Linq.XName> objects before you construct the XML tree, and then use the <xref:System.Xml.Linq.XName> objects instead of specifying strings for the element and attribute names. This technique can yield significant performance gains if you are creating a large number of elements (or attributes) with the same name.
+
+You should test pre-atomization with your scenario to decide if you should use it.
+
+## Example
+
+The following example demonstrates this:
+
+```vb
+Dim root1 As XName = "Root"
+Dim data As XName = "Data"
+Dim id As XName = "ID"
+
+Dim root2 As New XElement(root1, New XElement(data, New XAttribute(id, "1"), "4,100,000"), New XElement(data, New XAttribute(id, "2"), "3,700,000"), New XElement(data, New XAttribute(id, "3"), "1,150,000"))
```

Comment:
Let's break this line to avoid the scrollbar

```suggestion
Dim root2 As New XElement(root1, New XElement(data, New XAttribute(id, "1"), "4,100,000"),
                          New XElement(data, New XAttribute(id, "2"), "3,700,000"),
                          New XElement(data, New XAttribute(id, "3"), "1,150,000"))
```

![image](https://user-images.githubusercontent.com/12971179/73038129-2635b000-3e06-11ea-9ae4-1ef2f4059dc1.png)